redash/query_runner/athena.py

```diff
@@ -130,28 +135,41 @@ def __get_schema_from_glue(self):
         database_paginator = client.get_paginator('get_databases')
         table_paginator = client.get_paginator('get_tables')
 
-        for databases in database_paginator.paginate():
-            for database in databases['DatabaseList']:
-                iterator = table_paginator.paginate(DatabaseName=database['Name'])
-                for table in iterator.search('TableList[]'):
-                    table_name = '%s.%s' % (database['Name'], table['Name'])
-                    if table_name not in schema:
-                        column = [columns['Name'] for columns in table['StorageDescriptor']['Columns']]
-                        schema[table_name] = {'name': table_name, 'columns': column}
-                        for partition in table.get('PartitionKeys', []):
-                            schema[table_name]['columns'].append(partition['Name'])
+        query = 'DatabaseList[*]'
```

Comment:
I feel that this function is already involving several concerns, could you extract the concept of building the query to a helper function? Also, I think a balanced `if else` would signal the two different conditional branches here.

Reply:
Heck, might as well conceal the concept of the query here and just return the result of `database_paginator.paginate().search(...)`